server/routes/administration/non-sitting-days/non-sitting-days.controller.js

```diff
@@ -0,0 +1,174 @@
+
+        return res.render('administration/non-sitting-days.njk', {
+          postUrl,
+          holidayDates: holidayDates.response,
+          nonSittingDates: nonSittingDates,
+          holidayDateYears,
+          locationName: loggedInName.locationName,
+          bannerMessage,
+        });
+      } catch (err) {
+        app.logger.crit('Failed to fetch list of holidays and non eitting days', {
+          auth: req.session.authentication,
+          token: req.session.authToken,
+          error: typeof err.error !== 'undefined' ? err.error : err.toString(),
+        });
+
+        return res.render('_errors/generic.njk');
+      }
+
+    };
+  };
+
+  module.exports.postNonSittingDays = function(app) {
+    return async function(req, res) {
+
+      try {
+        return res.redirect(app.namedRoutes.build('administration.add-non-sitting-days.get'));
+      } catch (err) {
+
+        app.logger.crit('Failed eo add the non sitting day ', {
+          auth: req.session.authentication,
+          jwt: req.session.authToken,
+          error: (typeof err.error !== 'undefined') ? err.error : err.toString(),
+        });
+
+        return res.render('_errors/generic');
+
+      }
+    };
+  };
+
+  module.exports.getAddNonSittingDay = function(app) {
+    return async function(req, res) {
+      const tmpErrors = _.clone(req.session.errors);
+      const formFields = _.clone(req.session.formFields);
+
+      const cancelUrl = app.namedRoutes.build('administration.add-non-sitting-days.get');
+      const postUrl = app.namedRoutes.build('administration.add-non-sitting-days.post');
+
+      delete req.session.errors;
+      delete req.session.formFields;
+
+      try {
+        return res.render('administration/add-non-sitting-days.njk', {
+          cancelUrl,
+          postUrl,
+          errors: {
+            title: 'There is a problem',
+            count: typeof tmpErrors !== 'undefined' ? Object.keys(tmpErrors).length : 0,
+            items: tmpErrors,
+          },
+          formFields,
+        });
+      } catch (err) {
+        app.logger.crit('Failed ', {
+          auth: req.session.authentication,
+          token: req.session.authToken,
+          error: typeof err.error !== 'undefined' ? err.error : err.toString(),
+        });
+
+        return res.render('_errors/generic.njk');
+      }
+    };
+  };
+
+  module.exports.postAddNonSittingDay = function(app) {
+    return async function(req, res) {
+      var validatorResult;
+
+      validatorResult = validate(req.body, validator());
+      if (typeof validatorResult !== 'undefined') {
+        req.session.errors = validatorResult;
+        req.session.formFields = req.body;
+
+        return res.redirect(app.namedRoutes.build('administration.add-non-sitting-days.get'));
+      };
+
+      const payload = {
+        'date': req.body.nonSittingDate !== '' ? dateFilter(req.body.nonSittingDate, null, 'YYYY-MM-DD') : '',
```

Comment:
Should a missing nonSittingDate fail the validation? If it does, this check probably isn't needed.

Reply:
It is validated, I have removed the check.